src/daft-sql/src/functions.rs

```diff
@@ -301,17 +302,35 @@ impl SQLPlanner<'_> {
 
         fn get_func_from_sqlfunctions_registry(
             name: impl AsRef<str>,
-        ) -> SQLPlannerResult<Arc<dyn SQLFunction>> {
+        ) -> Option<Arc<dyn SQLFunction>> {
             let name = name.as_ref();
-            SQL_FUNCTIONS.get(name).cloned().ok_or_else(|| {
-                PlannerError::unsupported_sql(format!("Function `{}` not found", name))
-            })
+            SQL_FUNCTIONS.get(name).cloned()
+        }
+
+        fn get_func_from_session(
+            session: &Session,
+            name: impl AsRef<str>,
+        ) -> SQLPlannerResult<Option<Arc<dyn SQLFunction>>> {
+            let name = name.as_ref();
+
+            match session.get_function(name)? {
+                Some(f) => Ok(Some(Arc::new(f))),
+                None => Ok(None),
+            }
         }
 
         // lookup function variant(s) by name
         // SQL function names are case-insensitive
         let fn_name = func.name.to_string().to_lowercase();
-        let mut fn_match = get_func_from_sqlfunctions_registry(fn_name.as_str())?;
+
+        let mut fn_match =
+            if let Some(fn_match) = get_func_from_sqlfunctions_registry(fn_name.as_str()) {
+                fn_match
+            } else {
+                get_func_from_session(&self.context.borrow().session, &fn_name)?.ok_or_else(
+                    || PlannerError::unsupported_sql(format!("Function `{}` not found", fn_name)),
+                )?
```

Comment:
This is touching on a super interesting part of SQL which we don't have .. the  search_path. What I have been doing is searching attached tables first. The implication here is that a customer may now override a builtin function. I believe this is technically how it is supposed to be.

The current_schema's routines are by default first on the search_path, then the other schemas. In this case we could say our builtins. Technically there are no builtins 🤷  but it makes sense for us to always have builtins last to simulate what a full search_path would behave like. The important thing is checking attached things first.